### PR TITLE
OSDOCS-4653: Initial release notes file MicroShift

### PR DESCRIPTION
**Pull request by ShaunaDiaz**
[OSDOCS-4653](https://issues.redhat.com//browse/OSDOCS-4653): Initial release notes file MicroShift

Version(s):
4.12 only

Issue:
https://issues.redhat.com/browse/OSDOCS-4653

Link to docs preview:
https://54515--docspreview.netlify.app/microshift/latest/microshift_release_notes/microshift-4-12-release-notes.html

QE review:
N/A -- MicroShift is Developer Preview
SME approval is required 

Additional information:
These are the Dev Preview release notes.

### CODE REVIEW

**Comment by ShaunaDiaz** [comment]
@jogeo PTAL and let me know if lgtm or we need to make corrections.

**Comment by ShaunaDiaz** [comment]
3e2383284a adds in reference to Errata; any user who has Customer Portal access should be able to configure or search for Errata updates (even if they do not have support), per internal discussions with CCS and Docs teams.

File: microshift_release_notes/microshift-4-12-release-notes.adoc [hunk 1]
```diff
@@ -6,15 +6,71 @@ include::_attributes/attributes-microshift.adoc[]
...
+  to:
+    kind: Service
+    name: svc-1
+----
+
+* There is an issue resuming pods when a service is idled. Unidling is not supported by the {product-title} software, and manual re-scaling the replicas is required.
+
+* It is possible to over-provision the storage system by creating persistent volumes (PVs) that do not actually fit on the disk when using the `spare-gb` configuration value in `lvmd.conf`. To protect capacity from being used by the `topoLVM` driver, use a new separate volume group (VG) that is not exposed to topolvm.
+
+* There is a known issue involving `runc` and `systemd`. `Runc` is overly verbose about `mount ops` and `systemd` does not have the capability to filter out the messages. These messages are innocuous and do not indicate a problem with the system's operation.  This bug affects any Linux platform running a container runtime built on `runc`.
```
Comment (dhellmann) [inline microshift_release_notes/microshift-4-12-release-notes.adoc:47]:
```suggestion
* There is a known issue involving `runc` and `systemd`. `Runc` is overly verbose about `mount ops`. These messages are innocuous and do not indicate a problem with the system's operation.  This bug affects any Linux platform running a container runtime built on `runc`.
```

The next paragraph demonstrates how to set up a filter.

File: microshift_release_notes/microshift-4-12-release-notes.adoc [hunk 2]
```diff
@@ -6,15 +6,71 @@ include::_attributes/attributes-microshift.adoc[]
...
+[Mount] LogLevelMax=notice
+----
++
+And reload the daemon.
+
+* {product-title} uses long-lived certificates but some of the code imported from {ocp} prefers longer lifetimes and logs warning messages. The messages are innocuous and do not indicate a problem with the system's operation.
+
+* There is an issue with the way {product-title}'s CNI driver manages the firewall rules that means if the firewall settings are modified after {product-title} is started then {product-title} must be restarted. The recommended workaround is to configure firewall rules before starting {product-title}.
+
+* There is a known incompatibility in the Route API in {product-title} when compared with {ocp}. In {product-title}, the default value for the `spec.to.kind` and `spec.wildcardPolicy` fields are only applied if the API request does not include the field at all; an empty string is not replaced by the default value. The workaround is to ensure that all Routes have `spec.to.kind` and `spec.wildcardPolicy` set explicitly.
```
Comment (dhellmann) [inline microshift_release_notes/microshift-4-12-release-notes.adoc:62]: I think we fixed this, didn't we @benluddy ?

**Comment by ShaunaDiaz** [comment]
9c398a0 remove fixed Route API issue relnote

**Comment by ShaunaDiaz** [comment]
@dhellmann or @jogeo If either of you can PTAL and ready to label lgtm, that would be great.

File: microshift_release_notes/microshift-4-12-release-notes.adoc [hunk 3]
```diff
@@ -6,15 +6,69 @@ include::_attributes/attributes-microshift.adoc[]
...
+spec:
+  subdomain: svc-subdomain  ⇐ specify the subdomain
+  port:
+    targetPort: 8080
+  to:
+    kind: Service
+    name: svc-1
+----
+
+* There is an issue resuming pods when a service is idled. Unidling is not supported by the {product-title} software, and manual re-scaling the replicas is required.
```
Comment (mburke5678) [inline microshift_release_notes/microshift-4-12-release-notes.adoc:43]: Unidling [isn't actually a word](https://www.merriam-webster.com/dictionary/unidle), is it? I can't think of a good alternative, though.

File: microshift_release_notes/microshift-4-12-release-notes.adoc [hunk 4]
```diff
@@ -6,15 +6,69 @@ include::_attributes/attributes-microshift.adoc[]
...
+
+* It is possible to over-provision the storage system by creating persistent volumes (PVs) that do not actually fit on the disk when using the `spare-gb` configuration value in `lvmd.conf`. To protect capacity from being used by the `topoLVM` driver, use a new separate volume group (VG) that is not exposed to topolvm.
```
Comment (mburke5678) [inline microshift_release_notes/microshift-4-12-release-notes.adoc:45]: s/exposed to topolvm/exposed to the `topoLVM` driver ?? Or just `topoLVM`

File: microshift_release_notes/microshift-4-12-release-notes.adoc (hunk 3 above)
Comment (ShaunaDiaz) [reply microshift_release_notes/microshift-4-12-release-notes.adoc:43]: "Returning a service to an active state" perhaps?

File: microshift_release_notes/microshift-4-12-release-notes.adoc [hunk 5]
```diff
@@ -6,15 +6,69 @@ include::_attributes/attributes-microshift.adoc[]
...
+
+* There is a known issue involving `runc` and `systemd`. `Runc` is overly verbose about `mount ops`. These messages are innocuous and do not indicate a problem with the system's operation.  This bug affects any Linux platform running a container runtime built on `runc`.
```
Comment (mburke5678) [inline microshift_release_notes/microshift-4-12-release-notes.adoc:47]: I think `runc` is always lower-case `r`. Even at the start of the sentence. See https://github.com/opencontainers/runc

File: microshift_release_notes/microshift-4-12-release-notes.adoc [hunk 6]
```diff
@@ -6,15 +6,69 @@ include::_attributes/attributes-microshift.adoc[]
...
+You can suppress all `runc` mount logs through `systemd`. Create the file `/etc/systemd/system/run-containerd-.mount.d/10-silence.conf`, or whatever prefix you need to silence, with:
++
+[source, terminal]
+----
+[Mount] LogLevelMax=notice
+----
++
+And reload the daemon.
+
+* {product-title} uses long-lived certificates but some of the code imported from {ocp} prefers longer lifetimes and logs warning messages. The messages are innocuous and do not indicate a problem with the system's operation.
```
Comment (mburke5678) [inline microshift_release_notes/microshift-4-12-release-notes.adoc:58]: s/long-lived certificates but some/long-lived certificates. However, some

s/prefers/requires  (I think this qualifies as an [anthropomorphism](https://stylepedia.net/style/6.0/#Avoiding_Slang_Metaphors-MisleadingLanguage-Anthropomorphism), which I stumbled upon earlier today)

File: microshift_release_notes/microshift-4-12-release-notes.adoc [hunk 7]
```diff
@@ -6,15 +6,69 @@ include::_attributes/attributes-microshift.adoc[]
...
+
+* There is an issue with the way {product-title}'s CNI driver manages the firewall rules that means if the firewall settings are modified after {product-title} is started then {product-title} must be restarted. The recommended workaround is to configure firewall rules before starting {product-title}.
```
Comment (mburke5678) [inline microshift_release_notes/microshift-4-12-release-notes.adoc:60]: Something about _that means_ bugs me. Maybe _such that, if the firewall settings are modified..._

File: microshift_release_notes/microshift-4-12-release-notes.adoc (hunk 3 above)
Comment (dhellmann) [reply microshift_release_notes/microshift-4-12-release-notes.adoc:43]: We use the term "unidling" elsewhere in the docs. https://access.redhat.com/documentation/en-us/openshift_container_platform/4.1/html/applications/idling-applications#idle-unidling-applications_idling-applications

We should make sure it's clear here that services are not scaled back up automatically.

```suggestion
* There is an issue resuming pods when a service is idled. {product-title} does not support automatically reactivating services, and manual re-scaling the replicas is required.
```

File: microshift_release_notes/microshift-4-12-release-notes.adoc (hunk 4 above)
Comment (dhellmann) [reply microshift_release_notes/microshift-4-12-release-notes.adoc:45]: topoLVM is the default CSI driver. Saying just "topoLVM" makes sense in this context.

File: microshift_release_notes/microshift-4-12-release-notes.adoc (hunk 6 above)
Comment (dhellmann) [reply microshift_release_notes/microshift-4-12-release-notes.adoc:58]: I missed this one. We fixed it, so the note can come out. cc @pacevedom

File: microshift_release_notes/microshift-4-12-release-notes.adoc (hunk 3 above)
Comment (ShaunaDiaz) [reply microshift_release_notes/microshift-4-12-release-notes.adoc:43]: Thank you, @mburke5678 and @dhellmann, edit made.

File: microshift_release_notes/microshift-4-12-release-notes.adoc (hunk 6 above)
Comment (ShaunaDiaz) [reply microshift_release_notes/microshift-4-12-release-notes.adoc:58]: @mburke5678 You are right about that, though. So noted.

File: microshift_release_notes/microshift-4-12-release-notes.adoc [hunk 8]
```diff
@@ -6,15 +6,69 @@ include::_attributes/attributes-microshift.adoc[]
...
+
+[id="microshift-4-12-asynchronous-errata-updates"]
+== Asynchronous errata updates
+
+Security, bug fix, and enhancement updates for {product-title} {product-version} are released as asynchronous errata through the Red Hat Network. All {product-title} {product-version} errata is https://access.redhat.com/downloads/content/290/[available on the Red Hat Customer Portal]. See the https://access.redhat.com/support/policy/updates/openshift[{product-title} Life Cycle] for more information about asynchronous errata.
```
[the remark this answers is not in the document]
Comment (ShaunaDiaz) [reply microshift_release_notes/microshift-4-12-release-notes.adoc:65]: You are right, it's _erratum_ in singular. That is straight from OCP docs, unedited. Want to make a bug to fix that there as well? Fixed here.

**Pull request closed without merging**